lib/rspec/matchers/built_in/include.rb

```diff
@@ -134,6 +166,28 @@ def actual_collection_includes?(expected_item)
           actual.any? { |value| values_match?(expected_item, value) }
         end
 
+        if RUBY_VERSION < '1.9'
+          def count_enumerable(expected_item)
+            actual.select { |value| values_match?(expected_item, value) }.size
+          end
+        else
+          def count_enumerable(expected_item)
+            actual.count { |value| values_match?(expected_item, value) }
```

Comment:
It *has to be* more performant, since it doesn't build an intermediate array like `select{}.size` does, but I didn't measure it. It would depend on how slow `values_match?` is, but I'm pretty sure it's not major. Would you like me to make a PR with just the 1.8 compatible code?

Reply:
Anecdotally it's about 15-25% faster, leaving it is fine